diff -Naur binutils-2.28-original/bfd/config.bfd binutils-2.28/bfd/config.bfd
--- binutils-2.28-original/bfd/config.bfd	2017-03-02 10:23:53.000000000 +0200
+++ binutils-2.28/bfd/config.bfd	2025-12-17 20:48:43.317678582 +0200
@@ -413,6 +413,10 @@
     targ_underscore=no
     targ_cflags="-DARM_WINCE -DARM_COFF_BUGFIX"
     ;;
+  arm-*-pedigree*)
+    targ_defvec=arm_elf32_le_vec
+    targ_selvecs=arm_elf32_be_vec
+    ;;
   arm-*-pe*)
     targ_defvec=arm_pe_le_vec
     targ_selvecs="arm_pe_le_vec arm_pe_be_vec arm_pei_le_vec arm_pei_be_vec"
@@ -878,6 +882,13 @@
     targ_defvec=i386_elf32_vec
     targ_selfvecs="iamcu_elf32_vec i386chaos_vec"
     ;;
+#ifdef BFD64
+  amd64-*-pedigree* | \
+  x86_64-*-pedigree*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs=i386_elf32_vec
+    ;;
+#endif
 
   i860-*-mach3* | i860-*-osf1* | i860-*-coff*)
     targ_defvec=i860_coff_vec
diff -Naur binutils-2.28-original/binutils/configure binutils-2.28/binutils/configure
--- binutils-2.28-original/binutils/configure	2017-03-02 10:55:51.000000000 +0200
+++ binutils-2.28/binutils/configure	2025-12-17 20:48:43.707687342 +0200
@@ -14486,6 +14486,8 @@
 	  BUILD_WINDRES='$(WINDRES_PROG)$(EXEEXT)'
 	  BUILD_WINDMC='$(WINDMC_PROG)$(EXEEXT)'
 	  ;;
+  arm-*-pedigree*)
+    ;;
 	arm-*-pe*)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
diff -Naur binutils-2.28-original/binutils/configure.ac binutils-2.28/binutils/configure.ac
--- binutils-2.28-original/binutils/configure.ac	2017-03-02 10:23:53.000000000 +0200
+++ binutils-2.28/binutils/configure.ac	2025-12-17 20:48:43.709810297 +0200
@@ -306,6 +306,9 @@
 	  BUILD_WINDRES='$(WINDRES_PROG)$(EXEEXT)'
 	  BUILD_WINDMC='$(WINDMC_PROG)$(EXEEXT)'
 	  ;;
+    arm-*-pedigree*)
+      # Avoid pedigree OS target overlapping with arm-*-pe
+      ;;
 	arm-*-pe*)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
diff -Naur binutils-2.28-original/config.sub binutils-2.28/config.sub
--- binutils-2.28-original/config.sub	2016-12-23 10:40:17.000000000 +0200
+++ binutils-2.28/config.sub	2025-12-17 20:48:49.403543134 +0200
@@ -1541,6 +1541,8 @@
 		;;
 	-ios)
 		;;
+	-pedigree*)
+		;;
 	-none)
 		;;
 	*)
diff -Naur binutils-2.28-original/configure.ac binutils-2.28/configure.ac
--- binutils-2.28-original/configure.ac	2017-03-02 10:59:40.000000000 +0200
+++ binutils-2.28/configure.ac	2025-12-17 20:48:49.401148489 +0200
@@ -1828,7 +1828,7 @@
   # warn during gcc/ subconfigure; unless you're bootstrapping with
   # -flto it won't be needed until after installation anyway.
     case $target in
-      *-cygwin* | *-mingw* | *-apple-darwin* | *djgpp*) ;;
+      *-cygwin* | *-mingw* | *-apple-darwin* | *djgpp* | *pedigree*) ;;
       *) if test x"$enable_lto" = x"yes"; then
 	AC_MSG_ERROR([LTO support is not enabled for this target.])
         fi
diff -Naur binutils-2.28-original/gas/configure.tgt binutils-2.28/gas/configure.tgt
--- binutils-2.28-original/gas/configure.tgt	2017-03-02 10:23:53.000000000 +0200
+++ binutils-2.28/gas/configure.tgt	2025-12-17 20:48:46.649186195 +0200
@@ -163,6 +163,7 @@
   arm-*-pe)				fmt=coff em=pe ;;
   arm-*-riscix*)			fmt=aout em=riscix ;;
   arm-*-fuchsia*)			fmt=elf ;;
+  arm-*-pedigree*)		fmt=elf ;;
 
   avr-*-*)				fmt=elf bfd_gas=yes ;;
 
@@ -288,6 +289,7 @@
   i386-*-chaos)				fmt=elf ;;
   i386-*-rdos*)				fmt=elf ;;
   i386-*-darwin*)			fmt=macho ;;
+  i386-*-pedigree*)		fmt=elf ;;
 
   i860-*-*)				fmt=elf endian=little ;;
 
diff -Naur binutils-2.28-original/ld/configure.tgt binutils-2.28/ld/configure.tgt
--- binutils-2.28-original/ld/configure.tgt	2017-03-02 10:23:54.000000000 +0200
+++ binutils-2.28/ld/configure.tgt	2025-12-17 20:48:48.287790173 +0200
@@ -104,6 +104,7 @@
 arm-*-netbsd*)		targ_emul=armnbsd;
 			targ_extra_emuls="armelf armelf_nbsd armelfb_nbsd"  ;;
 arm-*-nto*)		targ_emul=armnto ;;
+arm-*-pedigree*)	targ_emul=pedigree_arm ;;
 arm-*-openbsd*)		targ_emul=armnbsd ;;
 arm-*-phoenix*)		targ_emul=armelf ;;
 arm-*-rtems*)		targ_emul=armelf ;;
@@ -247,6 +248,7 @@
 			targ_extra_emuls=elf_iamcu ;;
 x86_64-*-rdos*)		targ_emul=elf64rdos ;;
 x86_64-*-cloudabi*)	targ_emul=elf_x86_64_cloudabi ;;
+x86_64-*-pedigree*)	targ_emul=pedigree_x86_64 targ_extra_emuls="elf_i386" ;;
 i[3-7]86-*-bsd)		targ_emul=i386bsd ;;
 i[3-7]86-*-bsd386)	targ_emul=i386bsd ;;
 i[3-7]86-*-bsdi*)	targ_emul=i386bsd ;;
diff -Naur binutils-2.28-original/ld/Makefile.in binutils-2.28/ld/Makefile.in
--- binutils-2.28-original/ld/Makefile.in	2017-03-02 10:23:54.000000000 +0200
+++ binutils-2.28/ld/Makefile.in	2025-12-17 20:48:48.284456764 +0200
@@ -754,6 +754,8 @@
 	enios2linux.c \
 	epc532macha.c \
 	epdp11.c \
+	epedigree_arm.c \
+	epedigree_x86_64.c \
 	epjelf.c \
 	epjlelf.c \
 	eppclynx.c \
@@ -1456,6 +1458,8 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ens32knbsd.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/epc532macha.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/epdp11.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/epedigree_arm.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/epedigree_x86_64.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/epjelf.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/epjlelf.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eppclynx.Po@am__quote@
@@ -3289,6 +3293,12 @@
 epdp11.c: $(srcdir)/emulparams/pdp11.sh \
   $(srcdir)/emultempl/generic.em $(srcdir)/scripttempl/aout.sc ${GEN_DEPENDS}
 
+epedigree_arm.c: $(srcdir)/emulparams/pedigree_arm.sh \
+  $(ELF_DEPS) $(srcdir)/emultempl/armelf.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+
+epedigree_x86_64.c: $(srcdir)/emulparams/pedigree_x86_64.sh \
+  $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+
 epjelf.c: $(srcdir)/emulparams/pjelf.sh \
   $(ELF_GEN_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
