diff -Naur gcc-6.3.0-original/config.sub gcc-6.3.0/config.sub
--- gcc-6.3.0-original/config.sub	2016-08-31 11:49:03.000000000 +0300
+++ gcc-6.3.0/config.sub	2025-12-17 20:42:23.565414129 +0200
@@ -1531,6 +1531,9 @@
 		;;
 	-nacl*)
 		;;
+	-pedigree*)
+		os=-pedigree
+		;;
 	-none)
 		;;
 	*)
diff -Naur gcc-6.3.0-original/configure.ac gcc-6.3.0/configure.ac
--- gcc-6.3.0-original/configure.ac	2016-03-18 00:54:19.000000000 +0200
+++ gcc-6.3.0/configure.ac	2025-12-17 20:42:23.582081210 +0200
@@ -1067,6 +1067,7 @@
     noconfigdirs="$noconfigdirs target-libgloss"
     # this is not caught below because this stanza matches earlier
     case $target in
+      *-apple-darwin[[912]]* | *-cygwin* | *-mingw* | *djgpp* | *pedigree*) ;;
       or1k*-*-*) noconfigdirs="$noconfigdirs gdb" ;;
     esac
     ;;
@@ -1817,6 +1818,7 @@
   build_lto_plugin=yes
 ],[if test x"$default_enable_lto" = x"yes" ; then
     case $target in
+      *-apple-darwin[[912]]* | *-cygwin* | *-mingw* | *djgpp* | *pedigree*) ;;
       *-apple-darwin9* | *-cygwin* | *-mingw* | *djgpp*) ;;
       # On other non-ELF platforms, LTO has yet to be validated.
       *) enable_lto=no ;;
@@ -1828,7 +1830,8 @@
   # warn during gcc/ subconfigure; unless you're bootstrapping with
   # -flto it won't be needed until after installation anyway.
     case $target in
-      *-cygwin* | *-mingw* | *-apple-darwin* | *djgpp*) ;;
+      *-apple-darwin[[912]]* | *-cygwin* | *-mingw* | *djgpp* | *pedigree*) ;;
+      *-cygwin* | *-mingw* | *-apple-darwin* | *djgpp* | *pedigree*) ;;
       *) if test x"$enable_lto" = x"yes"; then
 	AC_MSG_ERROR([LTO support is not enabled for this target.])
         fi
@@ -1838,6 +1841,7 @@
   # Among non-ELF, only Windows platforms support the lto-plugin so far.
   # Build it unless LTO was explicitly disabled.
   case $target in
+      *-apple-darwin[[912]]* | *-cygwin* | *-mingw* | *djgpp* | *pedigree*) ;;
     *-cygwin* | *-mingw*) build_lto_plugin=$enable_lto ;;
     *) ;;
   esac
@@ -2577,6 +2581,7 @@
 
 alphaieee_frag=/dev/null
 case $target in
+      *-apple-darwin[[912]]* | *-cygwin* | *-mingw* | *djgpp* | *pedigree*) ;;
   alpha*-*-*)
     # This just makes sure to use the -mieee option to build target libs.
     # This should probably be set individually by each library.
diff -Naur gcc-6.3.0-original/gcc/config.gcc gcc-6.3.0/gcc/config.gcc
--- gcc-6.3.0-original/gcc/config.gcc	2016-11-07 23:38:43.000000000 +0200
+++ gcc-6.3.0/gcc/config.gcc	2025-12-17 20:42:14.728527349 +0200
@@ -904,6 +904,9 @@
     *) echo 'Unknown thread configuration for VxWorks'; exit 1 ;;
   esac
   ;;
+*-*-pedigree*)
+  use_gcc_stdint=provide
+  ;;
 *-*-elf|arc*-*-elf*)
   # Assume that newlib is being used and so __cxa_atexit is provided.
   default_use_cxa_atexit=yes
@@ -1117,10 +1120,17 @@
 	tm_file="dbxelf.h elfos.h arm/unknown-elf.h arm/elf.h arm/bpabi.h"
 	tmake_file="${tmake_file} arm/t-arm arm/t-arm-elf"
 	case ${target} in
-	arm*-*-eabi*)
+	arm*-*-eabi* | arm*-*-pedigree*)
 	  tm_file="$tm_file newlib-stdint.h"
 	  tmake_file="${tmake_file} arm/t-bpabi"
 	  use_gcc_stdint=wrap
+	  case ${target} in
+	  arm*-*-pedigree*)
+	    tm_file="$tm_file pedigree.h"
+	    tmake_file="${tmake_file} t-pedigree"
+	    use_gcc_stdint=provide
+	    ;;
+	  esac
 	  ;;
 	arm*-*-rtems*)
 	  tm_file="${tm_file} rtems.h arm/rtems.h newlib-stdint.h"
@@ -1603,6 +1613,11 @@
 	done
 	TM_MULTILIB_CONFIG=`echo $TM_MULTILIB_CONFIG | sed 's/^,//'`
 	;;
+x86_64-*-pedigree*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h i386/x86-64.h pedigree.h newlib-stdint.h"
+	tmake_file="i386/t-i386elf i386/t-crtstuff t-pedigree"
+	use_fixproto=yes
+	;;
 i[34567]86-pc-msdosdjgpp*)
 	xm_file=i386/xm-djgpp.h
 	tm_file="dbxcoff.h ${tm_file} i386/unix.h i386/bsd.h i386/gas.h i386/djgpp.h i386/djgpp-stdint.h"
diff -Naur gcc-6.3.0-original/gcc/Makefile.in gcc-6.3.0/gcc/Makefile.in
--- gcc-6.3.0-original/gcc/Makefile.in	2016-11-22 19:33:07.000000000 +0200
+++ gcc-6.3.0/gcc/Makefile.in	2025-12-17 20:42:14.721860516 +0200
@@ -230,6 +230,7 @@
 AWK = @AWK@
 CC = @CC@
 CXX = @CXX@
+CPP = @CPP@
 BISON = @BISON@
 BISONFLAGS =
 FLEX = @FLEX@
diff -Naur gcc-6.3.0-original/libgcc/config.host gcc-6.3.0/libgcc/config.host
--- gcc-6.3.0-original/libgcc/config.host	2016-11-19 05:03:55.000000000 +0200
+++ gcc-6.3.0/libgcc/config.host	2025-12-17 20:41:49.709060174 +0200
@@ -260,6 +260,10 @@
       ;;
   esac
   ;;
+*-*-pedigree*)
+  tmake_file="$tmake_file $cpu_type/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+  extra_parts="crtbegin.o crtend.o crtbeginS.o crtendS.o"
+  ;;
 *-*-rtems*)
   tmake_file="$tmake_file t-rtems"
   extra_parts="crtbegin.o crtend.o"
@@ -409,13 +413,19 @@
 	extra_parts="crtbegin.o crtend.o crti.o crtn.o"
 	unwind_header=config/arm/unwind-arm.h
 	;;
-arm*-*-eabi* | arm*-*-symbianelf* | arm*-*-rtems*)
+arm*-*-eabi* | arm*-*-symbianelf* | arm*-*-rtems* | arm*-*-pedigree*)
 	tmake_file="${tmake_file} arm/t-arm arm/t-elf t-fixedpoint-gnu-prefix"
 	tm_file="$tm_file arm/bpabi-lib.h"
 	case ${host} in
-	arm*-*-eabi* | arm*-*-rtems*)
+	arm*-*-eabi* | arm*-*-rtems* | arm*-*-pedigree*)
 	  tmake_file="${tmake_file} arm/t-bpabi t-crtfm"
 	  extra_parts="crtbegin.o crtend.o crti.o crtn.o"
+      case ${host} in
+      arm*-*-pedigree*)
+        # crti/crtn provided by musl
+        extra_parts="crtbegin.o crtend.o crtbeginS.o crtendS.o"
+        ;;
+      esac
 	  ;;
 	arm*-*-symbianelf*)
 	  tmake_file="${tmake_file} arm/t-symbian t-slibgcc-nolc-override"
@@ -595,6 +605,9 @@
 	tmake_file="${tmake_file} i386/t-dragonfly i386/t-crtstuff"
 	md_unwind_header=i386/dragonfly-unwind.h
 	;;
+x86_64-*-pedigree*)
+	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	;;
 i[34567]86-*-freebsd*)
 	tmake_file="${tmake_file} i386/t-freebsd i386/t-crtstuff"
 	md_unwind_header=i386/freebsd-unwind.h
diff -Naur gcc-6.3.0-original/libstdc++-v3/config/os/gnu-linux/os_defines.h gcc-6.3.0/libstdc++-v3/config/os/gnu-linux/os_defines.h
--- gcc-6.3.0-original/libstdc++-v3/config/os/gnu-linux/os_defines.h	2016-01-22 14:47:26.000000000 +0200
+++ gcc-6.3.0/libstdc++-v3/config/os/gnu-linux/os_defines.h	2025-12-17 20:41:51.266199759 +0200
@@ -38,13 +38,6 @@
 
 #include <features.h>
 
-// Provide a declaration for the possibly deprecated gets function, as
-// glibc 2.15 and later does not declare gets for ISO C11 when
-// __GNU_SOURCE is defined.
-#if __GLIBC_PREREQ(2,15) && defined(_GNU_SOURCE)
-# undef _GLIBCXX_HAVE_GETS
-#endif
-
 // Glibc 2.23 removed the obsolete isinf and isnan declarations. Check the
 // version dynamically in case it has changed since libstdc++ was configured.
 #define _GLIBCXX_NO_OBSOLETE_ISINF_ISNAN_DYNAMIC __GLIBC_PREREQ(2,23)
diff -Naur gcc-6.3.0-original/libstdc++-v3/configure.host gcc-6.3.0/libstdc++-v3/configure.host
--- gcc-6.3.0-original/libstdc++-v3/configure.host	2016-09-27 09:11:15.000000000 +0300
+++ gcc-6.3.0/libstdc++-v3/configure.host	2025-12-17 20:41:54.338016711 +0200
@@ -296,6 +296,10 @@
   openbsd*)
     os_include_dir="os/bsd/openbsd"
     ;;
+  pedigree*)
+    # Uses musl
+    os_include_dir="os/generic"
+    ;;
   qnx6.[12]*)
     os_include_dir="os/qnx/qnx6.1"
     c_model=c
diff -Naur gcc-6.3.0-original/libstdc++-v3/crossconfig.m4 gcc-6.3.0/libstdc++-v3/crossconfig.m4
--- gcc-6.3.0-original/libstdc++-v3/crossconfig.m4	2015-10-12 01:34:28.000000000 +0300
+++ gcc-6.3.0/libstdc++-v3/crossconfig.m4	2025-12-17 20:41:54.342134749 +0200
@@ -182,6 +182,9 @@
       AC_DEFINE(HAVE_ISNANL)
     fi
     ;;
+  *-pedigree*)
+    AC_CHECK_HEADERS([sys/types.h locale.h float.h])
+    ;;
   *-qnx6.1* | *-qnx6.2*)
     SECTION_FLAGS='-ffunction-sections -fdata-sections'
     AC_SUBST(SECTION_FLAGS) 
diff -Naur gcc-6.3.0-original/libtool.m4 gcc-6.3.0/libtool.m4
--- gcc-6.3.0-original/libtool.m4	2015-11-26 15:20:59.000000000 +0200
+++ gcc-6.3.0/libtool.m4	2025-12-17 20:42:23.575414378 +0200
@@ -5377,7 +5377,9 @@
   _LT_LINKER_SHLIBS($1)
   _LT_SYS_DYNAMIC_LINKER($1)
   _LT_LINKER_HARDCODE_LIBPATH($1)
-  LT_SYS_DLOPEN_SELF
+  if test "$cross_compiling" = no; then
+    LT_SYS_DLOPEN_SELF
+  fi
   _LT_CMD_STRIPLIB
 
   # Report which library types will actually be built
